transport/api/src/p2p.rs

```diff
@@ -567,7 +567,11 @@ pub async fn p2p_loop<T>(
                         let network = network.clone();
                         async_std::task::block_on(async move {
                             if !network.has(&peer_id).await {
-                                if let Err(e) = network.add(&peer_id, PeerOrigin::IncomingConnection, vec![]).await {
+                                let multi_address = match endpoint {
+                                    ConnectedPoint::Dialer { address, .. } => address,
+                                    ConnectedPoint::Listener { local_addr: _, send_back_addr } => send_back_addr,
+                                };
+                                if let Err(e) = network.add(&peer_id, PeerOrigin::IncomingConnection, vec![multi_address]).await {
```

Comment:
So what happens if the peer is not allowed in the registry?

Reply:
So you are suggesting to have an additional check and only update the multiaddress for peers that are included in the registry?

Reply:
Just asking. Can't remember, what happens with the banning behavior, but banned peers should likely not make it into the network.